Web/3DObject/src/WebAR3DObject.tsx

```diff
@@ -1,22 +1,28 @@
 import React, { createElement, useEffect, useState } from "react";
 import { WebAR3DObjectContainerProps } from "../typings/WebAR3DObjectProps";
-import { MeshComponent } from "../../../Shared/ComponentParent/src/MeshComponent";
-import { Mesh, Scene, SceneLoader, Texture } from "@babylonjs/core";
+import { MeshComponent, setAttributes } from "../../../Shared/ComponentParent/src/MeshComponent";
+import { Mesh, Scene, SceneLoader, Texture, Vector3 } from "@babylonjs/core";
 import "@babylonjs/loaders/glTF";
 import "@babylonjs/loaders/OBJ";
+import { GizmoComponent } from "../../../Shared/ComponentParent/src/GizmoComponent";
 
 export function WebAR3DObject(props: WebAR3DObjectContainerProps): React.ReactElement | void {
     const { mxMaterialTexture } = props;
     const [rootMesh, setRootMesh] = useState<Mesh | undefined>();
     const [allMeshes, setAllMeshes] = useState<Mesh[] | undefined>();
     const [scene, setScene] = useState<Scene>();
     const [texture, setTexture] = useState<Texture>();
+    const [rotationSet, setRotationSet] = useState<boolean>(false);
+    const [position, setPosition] = useState<Vector3>();
+    const [rotation, setRotation] = useState<Vector3>();
+    const [scale, setScale] = useState<Vector3>();
 
     useEffect(() => {
         if (mxMaterialTexture && scene) {
-            if (typeof(mxMaterialTexture.value) === "string") {
+            if (typeof mxMaterialTexture.value === "string") {
```

Comment:
It complains as `mxMaterialTexture.value` is probably never a string. I just checked types, and it looks like it is object with url property. So I would double check this place...

Reply:
It's look like there are WebImage and NativeImage.